qCC/ccCommandLineCommands.cpp

```diff
@@ -2688,6 +2690,164 @@ bool CommandMergeClouds::process(ccCommandLineInterface& cmd)
 	return true;
 }
 
+CommandSetGlobalShift::CommandSetGlobalShift()
+	: ccCommandLineInterface::Command(QObject::tr("Set global shift"), COMMAND_SET_GLOBAL_SHIFT)
+{}
+
+bool CommandSetGlobalShift::process(ccCommandLineInterface& cmd)
+{
+	cmd.print(QObject::tr("[SET GLOBAL SHIFT]"));
+
+	if (cmd.clouds().empty() && cmd.meshes().empty())
+	{
+		return cmd.error(QObject::tr("No loaded entity! (be sure to open one with \"-%1 [filename]\" before \"-%2\")").arg(COMMAND_OPEN, COMMAND_SET_GLOBAL_SHIFT));
+	}
+
+	//process globalshift options first
+	ccCommandLineInterface::GlobalShiftOptions globalShiftOptions;
+	cmd.processGlobalShiftCommand(globalShiftOptions);
+	//if it is not a valid global shift then an error msg already issued.
+	if (globalShiftOptions.mode != ccCommandLineInterface::GlobalShiftOptions::Mode::CUSTOM_GLOBAL_SHIFT)
+	{
+		return cmd.error(QObject::tr("Global shift must be in the form of three coordinates 'x' 'y' 'z'"));
+	}
+	CCVector3d newShift = globalShiftOptions.customGlobalShift;
+
+	//look for additional parameters
+	bool keepOrigFixed = false;
+	while (!cmd.arguments().empty())
```

Comment:
why the while loop? Since there can only be one option, you can remove it (and simply test whether the next argument (`cmd.arguments().front()`) is COMMAND_SET_GLOBAL_SHIFT_KEEP_ORIG_FIXED

Reply:
well it is done like this in several other command line functions, so i made it like it to keep the same structure. And it is easier to add more options if needed. But i can remove it.

Reply:
ahh that "else".., sorry missed that.